### ISSUE TITLE
Race condition testing

### ISSUE THREAD

**Issue by oze4** [gammazero/workerpool#38]
FYI

I noticed the following tests are inconsistent/broken..

 - `TestWaitingQueueSizeRace`
 - `TestStopRace`

```bash
# fails when count > ~30
go test -race -run ^TestWaitingQueueSizeRace$ -count=100

# this fails sometimes (run back to back to back, etc..)
go test -race -run ^TestWaitingQueueSizeRace$ -count=30

# without `-race` you need a high count, and run over and over
go test -run ^TestStopRace$ -count=2000

# this fails every time (I have yet to get it to pass)
go test -race -run ^TestStopRace$ -count=100

# this passes sometimes
go test -race -run ^TestStopRace$ -count=40
```

**Output:**

 - Example of how you need to run over and over to reveal inconsistencies

```
MY-COMPUTER:workerpool user$ go test -run ^TestStopRace$ -count=2000
PASS
ok      github.com/gammazero/workerpool 0.180s
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$ go test -run ^TestStopRace$ -count=2000
PASS
ok      github.com/gammazero/workerpool 0.177s
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$ go test -run ^TestStopRace$ -count=2000
--- FAIL: TestStopRace (0.00s)
    workerpool_test.go:331: Stop should not return in any goroutine
FAIL
exit status 1
FAIL    github.com/gammazero/workerpool 0.172s
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$ go test -run ^TestStopRace$ -count=2000
PASS
ok      github.com/gammazero/workerpool 0.168s
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$ go test -run ^TestStopRace$ -count=2000
--- FAIL: TestStopRace (0.00s)
    workerpool_test.go:331: Stop should not return in any goroutine
--- FAIL: TestStopRace (0.00s)
    workerpool_test.go:331: Stop should not return in any goroutine
FAIL
exit status 1
FAIL    github.com/gammazero/workerpool 0.169s
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$
MY-COMPUTER:workerpool user$
```

**Comment by gammazero** [comment]
@oze4 Thank you, nice catch.  The behavior of stop changed so that it releases paused tasks, which breaks the StopRace test.  I did not see it because it happened to pass in CI.

**Comment by gammazero** [comment]
Fixed in release v1.1.1

**Comment by oze4** [comment]
Nice! I'm using a couple of the race tests you wrote over in workerpoolxt, with proper credit, and someone else actually noticed it. 

That's when I decided to test your library to see if the root of the error was here or in workerpoolxt.

I forked this repo and took a stab at fixing it but...I was completely lost lol. I did notice the Pause addition but your code is still like 60% over my head tbh. It's been great learning from it, though. 

Cheers!
